### ISSUE TITLE
Issues with months and disabled dates

### ISSUE THREAD

**Issue by clnhll** [valor-software/ngx-bootstrap#1753]
There are two different issues related to disabled dates and months that are giving me some trouble:

1. Month is disabled in month picker if the first day of the month is disabled unless it's the current month.
<img width="265" alt="messages image 3577687316" src="https://cloud.githubusercontent.com/assets/12833388/23869530/436a6718-07f1-11e7-9a58-c01c1db62aef.png">
<img width="259" alt="messages image 1962804027" src="https://cloud.githubusercontent.com/assets/12833388/23869550/572b45ce-07f1-11e7-9b37-4f680c89abd8.png">

2. Selecting the current month using the month picker will select the first day of the month even if it is disabled. It should select the first non-disabled day in the month. 
<img width="253" alt="messages image 3767138548" src="https://cloud.githubusercontent.com/assets/12833388/23869610/98d24a9a-07f1-11e7-8032-9211763584b3.png">

Markup:
```html
<datepicker *ngIf="showDate"
                    [(ngModel)]="date"
                    [minDate]="minimumDate"
                    [maxDate]="maxDate"
                    [showWeeks]="false"
                    [dateDisabled]="disabledDates"
                    (ngModelChange)="changedDate()">
</datepicker>
```

**Comment by FabienDehopre** [comment]
I have the exact same issue.

**Comment by valorkin** [comment]
fixed in new datepicker
available in v1.9.2
http://valor-software.com/ngx-bootstrap/#/datepicker

**Comment by hardikchopra242** [comment]
Facing this issue in v7.1.2
Any other workarounds @clnhll @FabienDehopre
